docs/source/API/core/atomics/atomic_compare_exchange_strong.rst

```diff
@@ -1,33 +1,41 @@
 ``atomic_compare_exchange_strong``
 ==================================
 
+.. warning::
+   Deprecated since Kokkos 4.5,
+   use `atomic_compare_exchange <atomic_compare_exchange.html>`_ instead.
+
 .. role:: cppkokkos(code)
    :language: cppkokkos
 
-Header File: <Kokkos_Core.hpp>
+Defined in header ``<Kokkos_Atomic.hpp>`` which is included from ``<Kokkos_Core.hpp>``
 
 Usage
 -----
 
 .. code-block:: cpp
 
-   bool was_exchanged = atomic_compare_exchange_strong(ptr_to_value,
-						       comparison_value,
-						       new_value);
+   bool was_exchanged = atomic_compare_exchange_strong(&obj, expected, desired);
+   //                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   //                   deprecated since Kokkos 4.5
+   //                   instead prefer
+   //                   expected == atomic_compare_exchange(&obj, expected, desired)
```

Comment:
Would using the [`deprecated` directive](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-deprecated) be more straight forward?

Reply:
I used the "warning" directive above (L4)
I find the rendering not great with our current configuration (italic "Deprecated since version X.Y:" followed by  explanation)
I suppose this is a bit redundant.  My intention was that it is harder to miss that it is deprecated